client/components/Main/CreateNewSubnavDropdown.tsx

```diff
@@ -16,14 +16,42 @@ interface IProps {
     privileges: {[key: string]: number};
     createEventFromTemplate(template: IEventTemplate): void;
     eventTemplates: Array<IEventTemplate>;
+    recentTemplatesId: Array<string>;
 }
 
 class CreateNewSubnavDropdownFn extends React.PureComponent<IProps> {
+    constructor(props: IProps) {
+        super(props);
+        this.getRecentTemplates = this.getRecentTemplates.bind(this);
+    }
+
+    getRecentTemplates() {
+        const {recentTemplatesId, eventTemplates} = this.props;
+
+        if (recentTemplatesId.length !== 0) {
+            return eventTemplates.filter((template) =>
+                recentTemplatesId.includes(template._id)
+            );
+        }
+        return [];
+    }
     render() {
         const {gettext} = superdeskApi.localization;
-        const {addEvent, addPlanning, createPlanningOnly, privileges, createEventFromTemplate} = this.props;
+        const {
+            addEvent,
+            addPlanning,
+            createPlanningOnly,
+            privileges,
+            createEventFromTemplate,
+            recentTemplatesId,
+            eventTemplates
+        } = this.props;
         const items: Array<IDropdownItem> = [];
 
+        const recentTemplates = this.getRecentTemplates().sort(
```

Comment:
Why are the recent templates sorted? I would have though the sort order would be the order of last used to least used

Reply:
to maintain this order **last used to least used**, we used sort here.